interfaces/builtin/home.go

```diff
@@ -63,14 +76,56 @@ owner /run/user/[0-9]*/gvfs/{,**} r,
 owner /run/user/[0-9]*/gvfs/*/**  w,
 `
 
+const homeConnectedPlugAppArmorWithAllRead = `
+# Allow non-owner read to non-hidden and non-snap files and directories
+@{HOME}/               r,
+@{HOME}/[^s.]**        r,
+@{HOME}/s[^n]**        r,
+@{HOME}/sn[^a]**       r,
+@{HOME}/sna[^p]**      r,
+@{HOME}/snap[^/]**     r,
+@{HOME}/{s,sn,sna}{,/} r,
+`
+
+type homeInterface struct {
+	commonInterface
+}
+
+func (iface *homeInterface) BeforePreparePlug(plug *snap.PlugInfo) error {
+	// It's fine if 'read' isn't specified, but if it is, it needs to be
+	// either 'all' or 'owner'
+	if r, ok := plug.Attrs["read"]; ok {
+		_, ok = r.(string)
+		if !ok || (r != "all" && r != "owner") {
+			return fmt.Errorf(`home plug requires the "read" be either 'all' or 'owner'`)
+		}
+	}
+
+	return nil
+}
+
+func (iface *homeInterface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	var read string
+	_ = plug.Attr("read", &read)
```

Comment:
Could we use plug.Attr in the BeforePreparePlug above?

Reply:
I'm doing it this way in BeforePreparePlug because I want to see if it exists at all (since it is ok if it doesn't) and if it does make sure it is a string. I could just ignore if it isn't a string, but this isn't what we do elsewhere (eg browser-support, content and docker-support). The reason I can use plug.Attr() in AppArmorConnectedPlug is because by the time it is run, I'm sure things are ok and if read isn't an attribute, then the 'read' variable is simply empty and won't match `read != "all"`. As such, I prefer to leave the logic as is. If people feel strongly about changing it, we should change it in all of the interfaces that do similar checks.